### PR TITLE
Feature/704 generate store resolve cah

### PR DESCRIPTION
**Pull request by Kasshern**
# Description

Leaping off of existing PR branches this PR adds in an extra leafValue variable within the BpiAccountStateTreeLeafValue schema. 
This leaf value is the content addressable hash that corresponds to a state controller endpoint introduced here. When queried that endpoint currently returns the state tree, and all leaf related information (i.e. leaf index, value, merkelized payload, and witness)
## Related Issue
#704 

## Motivation and Context

Required for purposes of external retrieval and verification of state

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Request to be added as a Code Owner/Maintainer

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] I commit to abide by the Responsibilities of Code Owners/Maintainers.

### CODE REVIEW

**Comment by Therecanbeonlyone1969** [comment]
@Kasshern do you want me to review this PR now, or do you want to wait until #734 and #735 are merged, and you have resolved all conflicts?

**Comment by Kasshern** [comment]
> @Kasshern do you want me to review this PR now, or do you want to wait until #734 and #735 are merged, and you have resolved all conflicts?

Conflicts have been rebased. I would still wait until the others are merged because it branches from them, so you will see a lot redundant content.

File: examples/bri-3/package.json
Comment (ognjenkurtic) [inline examples/bri-3/package.json:44]: Something is strange. Why is this still shown as a change after rebase main? Main already has prisma/client 5.2.0  @Kasshern

Comment (Kasshern) [reply examples/bri-3/package.json]: Yeah something was strange. Perhaps because I branched from another branch that wasn't rebased? unsure. Anyways looks better now. I think...

**Review by biscuitdey — APPROVED** [review]
LGTM

File: examples/bri-3/prisma/migrations/20230821213120_add_bpi_account_state_tree_storage/migration.sql
Comment (ognjenkurtic) [inline examples/bri-3/prisma/migrations/20230821213120_add_bpi_account_state_tree_storage/migration.sql:11]: this should not exist as it was just introduce while trying out the path approach for merkle tree storage. it does not exist in prisma.schema

File: examples/bri-3/prisma/migrations/20230821213120_add_bpi_account_state_tree_storage/migration.sql
Comment (ognjenkurtic) [inline examples/bri-3/prisma/migrations/20230821213120_add_bpi_account_state_tree_storage/migration.sql:8]: there is already a migration on main that executes this

File: examples/bri-3/src/bri/identity/bpiAccounts/agents/bpiAccountsStorage.agent.ts
Comment (ognjenkurtic) [inline examples/bri-3/src/bri/identity/bpiAccounts/agents/bpiAccountsStorage.agent.ts:30]: this comment was removed on main

File: examples/bri-3/src/bri/state/agents/state.agent.ts
Comment (ognjenkurtic) [inline examples/bri-3/src/bri/state/agents/state.agent.ts:13]: these two imports are not used?

File: examples/bri-3/src/bri/state/api/state.controller.ts
Comment (ognjenkurtic) [inline examples/bri-3/src/bri/state/api/state.controller.ts:10]: getLeafValueContent

File: examples/bri-3/src/bri/state/capabilities/getStateContent/getStateContentQuery.handler.ts
Comment (ognjenkurtic) [inline examples/bri-3/src/bri/state/capabilities/getStateContent/getStateContentQuery.handler.ts:33]: why are we returninga domain object and not a DTO here?

File: examples/bri-3/src/bri/state/models/stateContent.ts
Comment (ognjenkurtic) [inline examples/bri-3/src/bri/state/models/stateContent.ts:5]: what is the difference between StateContent domain object and StateLeafValues domain object? Can we one instead of both?

Comment (ognjenkurtic) [reply examples/bri-3/src/bri/state/models/stateContent.ts:5]: If it is to much work we can also merge with both of these and tackle this comment laters as part of mil5

File: examples/bri-3/prisma/migrations/20230821213120_add_bpi_account_state_tree_storage/migration.sql
Comment (Kasshern) [reply examples/bri-3/prisma/migrations/20230821213120_add_bpi_account_state_tree_storage/migration.sql]: removed.

File: examples/bri-3/prisma/migrations/20230821213120_add_bpi_account_state_tree_storage/migration.sql
Comment (Kasshern) [reply examples/bri-3/prisma/migrations/20230821213120_add_bpi_account_state_tree_storage/migration.sql]: removed

File: examples/bri-3/src/bri/identity/bpiAccounts/agents/bpiAccountsStorage.agent.ts
Comment (Kasshern) [reply examples/bri-3/src/bri/identity/bpiAccounts/agents/bpiAccountsStorage.agent.ts]: removed

File: examples/bri-3/src/bri/state/agents/state.agent.ts
Comment (Kasshern) [reply examples/bri-3/src/bri/state/agents/state.agent.ts]: removed

File: examples/bri-3/src/bri/state/api/state.controller.ts
Comment (Kasshern) [reply examples/bri-3/src/bri/state/api/state.controller.ts]: change naming scheme for all to stateTreeLeafValueContent

File: examples/bri-3/src/bri/state/capabilities/getStateContent/getStateContentQuery.handler.ts
Comment (Kasshern) [reply examples/bri-3/src/bri/state/capabilities/getStateContent/getStateContentQuery.handler.ts]: This one was a trick to see if you were paying attention.

File: examples/bri-3/src/bri/state/models/stateContent.ts
Comment (Kasshern) [reply examples/bri-3/src/bri/state/models/stateContent.ts]: no difference now. One had the state tree the other did not. Switched to a single class called StateTreeLeafValueContent

**Review by ognjenkurtic — APPROVED** [review]

File: examples/bri-3/src/bri/state/state.profile.ts
Comment (Therecanbeonlyone1969) [inline examples/bri-3/src/bri/state/state.profile.ts:15]: shouldn't one of them be `StateTreeLeafValueContentDto`?

**Review by Therecanbeonlyone1969 — COMMENTED** [review]
Left a question, once resolved I will approve

File: examples/bri-3/src/bri/state/state.profile.ts
Comment (ognjenkurtic) [reply examples/bri-3/src/bri/state/state.profile.ts:15]: Nope, because of this issue https://github.com/eea-oasis/baseline/issues/740. Will be handled as part of mil5.

Comment (ognjenkurtic) [reply examples/bri-3/src/bri/state/state.profile.ts:15]: @Therecanbeonlyone1969 @biscuitdey  please reapprove

File: examples/bri-3/src/bri/state/agents/state.agent.ts
Comment (biscuitdey) [inline examples/bri-3/src/bri/state/agents/state.agent.ts:29]: merkleTreetStorageAgent (Mis-spelt) --> merkleTreeStorageAgent

**Review by biscuitdey — APPROVED** [review]
The misspelt word can be changed in the next PR. Approving for now.

File: examples/bri-3/src/bri/state/agents/state.agent.ts
Comment (Kasshern) [reply examples/bri-3/src/bri/state/agents/state.agent.ts:29]: Good catch. Its because its a Treat + tree = Treet. We will change in one of the next PRs.

**Pull request merged**
